metadata-io/src/main/java/com/linkedin/metadata/timeline/eventgenerator/EntityChangeEventGenerator.java

```diff
@@ -0,0 +1,45 @@
+package com.linkedin.metadata.timeline.eventgenerator;
+
+import com.datahub.authentication.Authentication;
+import com.github.fge.jsonpatch.JsonPatch;
+import com.linkedin.common.AuditStamp;
+import com.linkedin.common.urn.Urn;
+import com.linkedin.data.template.RecordTemplate;
+import com.linkedin.entity.client.EntityClient;
+import com.linkedin.metadata.entity.EntityAspect;
+import com.linkedin.metadata.entity.EntityService;
+import com.linkedin.metadata.timeline.data.ChangeCategory;
+import com.linkedin.metadata.timeline.data.ChangeEvent;
+import com.linkedin.metadata.timeline.data.ChangeTransaction;
+import java.util.List;
+import javax.annotation.Nonnull;
+
+
+public abstract class EntityChangeEventGenerator<T extends RecordTemplate> {
```

Comment:
I thought we'd discussed making the generator "Declare" which aspect types it can work with? Did we can this idea?

Reply:
Discussed offline, will add a TODO to do this later